cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -314,7 +315,35 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 			if impl.AWSBucket != "" {
 				bucket = impl.AWSBucket
 			}
-			_, err = a.Upload(path.Join(outputDirectory, exportPath, options.Filename), bucket, key)
+
+			imagePath := path.Join(outputDirectory, exportPath, options.Filename)
+
+			// *** SPECIAL VMDK HANDLING START ***
+			// Upload the VMDK image as stream-optimized.
+			// The VMDK conversion is applied only when the job was submitted by Weldr API,
```

Comment:
Note that this comment isn't correct, `args.StreamOptimized` is now also passed by the cloud API.

Reply:
It is, but the image is converted only if the `args.ImageName` is not empty and only Weldr API is setting it... So the comment is actually correct, but the reason why it is so is not obvious... 😕 
https://github.com/osbuild/osbuild-composer/blob/c9a08da29fdb06f1bdabf4582d9f08ed6398cc97/cmd/osbuild-worker/jobimpl-osbuild.go#L193-L201

Reply:
Ah, I misread the comment, sorry! Of course, it's correct. :)